src/cdf.rs

```diff
@@ -5653,7 +5653,7 @@ pub unsafe fn rav1d_cdf_thread_copy(dst: *mut CdfContext, src: *const CdfThreadC
 }
 
 pub unsafe fn rav1d_cdf_thread_alloc(
-    c: *mut Rav1dContext,
+    c: &Rav1dContext,
```

Comment:
When you make these things into references, can you also update all of the `(*c)`s in the `fn` to just `c`s?  It's not strictly necessary, but it makes reading the `fn` and cleaning it up further a lot easier.

Reply:
I was thinking we should do that in a separate commit to avoid extra noise for the more important review. Not sure it matters a ton whether it's in this change or not, I don't have a strong preference for future changes. Maybe it's easy enough to just read the deref changes if we're making enough other changes in the function body (but we aren't here).

Reply:
I think if we can make the ptr to ref change first, that's ideal (e.x. `*mut` => `&mut` and `(*c)` => `c`, which we can here).  If the `*mut` => `&` change is only possible (and not the `*mut => &mut` change) because of the rest of the PR, then it makes sense to do it together in the same PR, or save the `(*c)` => `c` change to a follow-up commit.  That way it's also clear which changes are specifically only possible due to the other changes in a PR (e.x. the ptr to ref change here is not).

Reply:
Done in #687.